tests/test_teacher_views.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+from portal.student_views import view_schedule
+from .test_courses import login, logout
+
+def test_all_grades(client):
+    # Make sure anonymous users cannot access
+    assert client.get('/course/216/session/1/all_grades').status_code == 302
+    # Login
+    rv = login(client, 'teacher@stevenscollege.edu', 'qwerty')
+    assert b'Logged in' in rv.data
+    # Make sure teacher can access
+    assert client.get('/course/216/session/1/all_grades').status_code == 403
+    # Log out of teacher who should not have access
+    rv = logout(client)
+    assert b'TSCT Portal Login' in rv.data
+
+    rv = login(client, 'teacher2@stevenscollege.edu', 'PASSWORD')
+    assert b'Logged in' in rv.data
+    # should be able  to access
+    assert client.get('/course/216/session/1/all_grades').status_code == 200
+
+    response = client.get('/course/216/session/1/all_grades')
+    # Making sure students grades are displaying
+    assert b'bob phillp' in response.data
+    assert b'F' in response.data
```

Comment:
In this specific instance this is completely fine, but keep in mind that a vague assert like `F` could pass when it shouldn't.  For example, if one of the students was named 'Frank' or the session name was 'ENG-216-F'.  
It may be better to check for `<td>F</td>` or something similar.